### PR TITLE
Adding ansys-pythonnet recipe

### PR DESCRIPTION
**Pull request by SMoraisAnsys**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by SMoraisAnsys** [comment]
@koubaa As asked a few month ago (yeah sorry for that) I'll continue to work on providing recipe for some of our packages. Can you confirm that you are willing to be listed as a maintainer ?

**Comment by koubaa** [comment]
@SMoraisAnsys confirmed

**Comment by SMoraisAnsys** [comment]
@conda-forge/python, ready for review!

**Comment by SMoraisAnsys** [comment]
@conda-forge/help-python, ready for review!

File: recipes/ansys-pythonnet/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/ansys-pythonnet-{{ version }}.tar.gz
+  sha256: 369a0a5a838a0991f755b6d63c319ab6997f9dc464d016187227be5cd860a9cb
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  build:
+    - dotnet ~=7.0
```
Comment (SMoraisAnsys) [inline recipes/ansys-pythonnet/meta.yaml:18]:
```suggestion
    - mono
    - dotnet ~=7.0
```

File: recipes/ansys-pythonnet/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  host:
+    - python
+    - setuptools >=61
+    - wheel
+    - pip
+  run:
+    - python
+    - clr_loader >=0.2.6,<0.3.0
+    - dotnet ~=7.0
```
Comment (SMoraisAnsys) [inline recipes/ansys-pythonnet/meta.yaml:27]:
```suggestion
    - mono
    - dotnet ~=7.0
```

File: recipes/ansys-pythonnet/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  host:
+    - python
+    - setuptools >=61
+    - wheel
+    - pip
+  run:
+    - python
+    - clr_loader >=0.2.6,<0.3.0
+    - mono
```
Comment (SMoraisAnsys) [inline recipes/ansys-pythonnet/meta.yaml:28]:
```suggestion
    - mono  # [linux, osx]
```

File: recipes/ansys-pythonnet/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/ansys-pythonnet-{{ version }}.tar.gz
+  sha256: 369a0a5a838a0991f755b6d63c319ab6997f9dc464d016187227be5cd860a9cb
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  build:
+    - mono
```
Comment (SMoraisAnsys) [inline recipes/ansys-pythonnet/meta.yaml:18]:
```suggestion
    - mono  # [linux, osx]
```

File: recipes/ansys-pythonnet/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
+{% set name = "ansys-pythonnet" %}
+{% set version = "3.1.0rc3" %}
```
Comment (ocefpaf) [inline recipes/ansys-pythonnet/meta.yaml:2]: This one falls under https://github.com/conda-forge/cfep/blob/main/cfep-05.md

File: recipes/ansys-pythonnet/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  sha256: 369a0a5a838a0991f755b6d63c319ab6997f9dc464d016187227be5cd860a9cb
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  build:
+    - mono  # [not win]
+    - dotnet ~=7.0
```
Comment (ocefpaf) [inline recipes/ansys-pythonnet/meta.yaml:19]: Are these build dependencies? Should they be in host instead?

**Comment by SMoraisAnsys** [comment]
@ocefpaf Thanks for reviewing this PR, I'll address your comments ASAP !

File: recipes/ansys-pythonnet/meta.yaml (hunk 5 above)
Comment (SMoraisAnsys) [reply recipes/ansys-pythonnet/meta.yaml:2]: I've added the required change, thanks for pointing that out!

File: recipes/ansys-pythonnet/meta.yaml (hunk 6 above)
Comment (SMoraisAnsys) [reply recipes/ansys-pythonnet/meta.yaml:19]: Yes they are build dependencies.

**Comment by SMoraisAnsys** [comment]
@ocefpaf Thanks again for the feedback, I've updated the recipe. Sorry for the latency, holiday and stacked up work prevented me from performing those minor changes.

File: recipes/ansys-pythonnet/meta.yaml (hunk 6 above)
Comment (ocefpaf) [reply recipes/ansys-pythonnet/meta.yaml:19]: Note that host is "build" and build are compilers and other things. I know,  confusing, but b/c you don't seem to have a compiled extension, everything used to "build" should be in host.

Comment (ocefpaf) [reply recipes/ansys-pythonnet/meta.yaml:19]: @SMoraisAnsys please check this. As far as my tests go you do not need these in the build section.

Comment (SMoraisAnsys) [reply recipes/ansys-pythonnet/meta.yaml:19]: @ocefpaf Thanks for the reminder, I didn't see your previous answer, I'll double check and clarify this in my head. Thanks for your time and care !

File: recipes/ansys-pythonnet/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  host:
+    - python
+    - setuptools >=61
+    - wheel
+    - pip
```
Comment (SMoraisAnsys) [inline recipes/ansys-pythonnet/meta.yaml:24]:
```suggestion
  host:
    - dotnet ~=7.0
    - mono  # [not win]
    - python
    - setuptools >=61
    - wheel
    - pip
```

**Comment by SMoraisAnsys** [comment]
@ocefpaf Thanks again for your guidance and care !

**Pull request merged**